jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt

```diff
@@ -34,28 +41,114 @@ class DownloadObjectAction(private val project: Project, treeTable: S3TreeTable)
         }
     }
 
-    override fun enabled(nodes: List<S3TreeNode>): Boolean = nodes.all { it is S3TreeObjectNode }
+    private fun downloadSingle(project: Project, file: S3TreeObjectNode) {
+        val destination = getDownloadLocation(false) ?: return
+
+        val downloadFile = if (destination.isDirectory()) {
+            destination.resolve(file.name)
```

Comment:
What does this do?

Reply:
If you select folder foo, we put the file name on it, to make foo/bar

Reply:
I should rename var to `destinationFile` and `selectedDefintion` above